src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -404,11 +405,12 @@ Command NotebookParser()
             var notebookParserCommand = new Command(
                 "notebook-parser",
                 LocalizationResources.Cli_dotnet_interactive_notebook_parserDescription());
-            notebookParserCommand.Handler = CommandHandler.Create(async () =>
+            notebookParserCommand.Handler = CommandHandler.Create(async (InvocationContext context) =>
             {
                 Console.InputEncoding = Encoding.UTF8;
                 Console.OutputEncoding = Encoding.UTF8;
                 var notebookParserServer = new NotebookParserServer(Console.In, Console.Out);
+                context.GetCancellationToken().Register(() => notebookParserServer.Dispose());
```

Comment:
Is it possible to test this?

Reply:
Most of the implementation details are already tested in System.CommandLine. The final composed functionality involves both VS Code and .NET Interactive together, and is OS-specific.

We're planning to publish a VS Code Insiders release and seeing if this resolves the issue. Once we think this is the right fix, we'll look at how to write an effective pinning test.